t/t-clone.sh

```diff
@@ -285,16 +359,48 @@ begin_test "clone ClientCert with homedir certs"
   ]" | lfstest-testutils addcommits
 
   git push origin main
+
+  # Now clone again with 'git lfs clone', test specific clone dir
+  # Test with both unencrypted and encrypted client certificate keys
+  cd "$TRASHDIR"
+
+  for enc in "false" "true"; do
+    if [ "$enc" = "true" ]; then
+      git config --global "http.$LFS_CLIENT_CERT_URL/.sslKey" "~/lfs-client-key-file-encrypted"
+      git config --global "http.sslCertPasswordProtected" "$enc"
+    fi
+
+    newclonedir="${reponame}-${enc}"
+    git lfs clone "$CLIENTCERTGITSERVER/$reponame" "$newclonedir" 2>&1 | tee lfsclone.log
+    grep "Cloning into" lfsclone.log
+    grep "Downloading LFS objects:" lfsclone.log
+    # should be no filter errors
+    grep "filter" lfsclone.log && exit 1
+    grep "error" lfsclone.log && exit 1
+    # should be cloned into location as per arg
+    [ -d "$newclonedir" ]
+
+    # check a few file sizes to make sure pulled
+    pushd "$newclonedir"
+      [ $(wc -c < "file1.dat") -eq 100 ]
+      [ $(wc -c < "file2.dat") -eq 75 ]
+      [ $(wc -c < "file3.dat") -eq 30 ]
+      assert_hooks "$(dot_git_dir)"
+      [ ! -e "lfs" ]
+      assert_clean_status
+    popd
+
+    # Now check clone with standard 'git clone' and smudge download
+    rm -rf "$reponame"
+    git clone "$CLIENTCERTGITSERVER/$reponame"
```

Comment:
Shouldn't we check that `git clone` creates files in `$reponame` that have the expected names and sizes as we did above for `git lfs clone`?

Reply:
That's a good question.  This particular test design, with a `git clone` [command](https://github.com/git-lfs/git-lfs/blob/f1249933f1e33b758743c6cfdf02e473f2935b07/t/t-clone.sh#L160-L162) after the `git lfs clone` command and subsequent checks, dates back to commits  4c64e8212454f248478f2d431f3e6931e3b67fe6 and 8f91a1bc7fe0c74cde254b432302e6ac871d1a86 of PR #1067, when the `cloneSSL` test was introduced.  It was then replicated in the `clone ClientCert` test when that was added in commit daba49ae43340ee30b2e7c9a6b8d5927dbd71f51 of PR #1893, and now this PR replicates it again into this `clone ClientCert with homedir certs` test.

In all of these cases, we're just relying on the `set -e` shell option to cause the tests to fail if the `git clone` doesn't succeed.  We definitely have some other tests, like those in the `t/t-filter-process.sh` script, which [confirm](https://github.com/git-lfs/git-lfs/blob/f1249933f1e33b758743c6cfdf02e473f2935b07/t/t-filter-process.sh#L44-L63) Git LFS objects have been fetched by `git clone`.

Still, we might as well flesh out our tests in `t/t-clone.sh` with some additional checks, just to be on the safe side.  I'll push a commit which does that to this PR, and then merge.  Thank you for the speedy review!